src/main/scala/com/nec/vectorengine/VeProcessImpl.scala

```diff
@@ -0,0 +1,221 @@
+package com.nec.vectorengine
+
+import com.nec.colvector.{VeColVectorSource => VeSource}
+import scala.util.Try
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Path}
+import com.typesafe.scalalogging.LazyLogging
+import org.bytedeco.javacpp.{BytePointer, LongPointer, Pointer}
+import org.bytedeco.veoffload.global.veo
+import org.bytedeco.veoffload.{veo_args, veo_proc_handle, veo_thr_ctxt}
+
+final case class WrappingVeo private (val node: Int,
+                                      identifier: String,
+                                      handle: veo_proc_handle,
+                                      tcontext: veo_thr_ctxt)
+                                      extends VeProcess with LazyLogging {
+
+  implicit class ExtendedPointer(buffer: Pointer) {
+    def nbytes: Long = {
+      buffer.limit * buffer.sizeof
+    }
+  }
+
+  logger.info(s"Opened VE process (Node ${node}) @ ${handle.address}: ${handle}")
+  logger.info(s"Opened VEO asynchronous context @ ${tcontext.address}: ${tcontext}")
+  logger.info(s"VEO version ${version}; API version ${apiVersion}")
+
+  private var opened = true
+
+  private[vectorengine] def requireValidBuffer(buffer: Pointer): Unit = {
+    require(buffer.address > 0, s"Buffer has an invalid address ${buffer.address}; either it is un-initialized or already closed")
+    require(buffer.nbytes > 0, s"Buffer has a declared size of ${buffer.nbytes}")
+  }
+
+  def isOpen: Boolean = {
+    opened
+  }
+
+  lazy val apiVersion: Int = {
+    veo.veo_api_version
+  }
+
+  lazy val version: String = {
+    val bytes = veo.veo_version_string
+    bytes.getString(StandardCharsets.US_ASCII)
+  }
+
+  lazy val source: VeSource = {
+    VeSource(identifier)
+  }
+
+  def allocate(size: Long): Long = {
+    require(size > 0L, s"Requested size ${size} is invalid")
+    // Value is initialized to 0
+    val ptr = new LongPointer(1)
+    val result = veo.veo_alloc_mem(handle, ptr, size)
+    val location = ptr.get
+    require(result == 0, s"Memory allocation failed with code: ${result}")
+    require(location > 0, s"Memory allocation returned an invalid address: ${ptr.get}")
+    logger.trace(s"Allocated ${size} bytes ==> ${ptr}")
+    ptr.close
+    location
+  }
```

Comment:
I think having something that (optionally) registers allocations and can complain when they aren't freed is an actually useful feature of the old way of doing things. 

I agree that all of the implicits that it used to do that are not quite as great though.

Reply:
That is a good point; let me update the PR with this.  I'm thinking of keeping the allocation tracker separate from the allocations metric (currently both are packed together in `ProcessExecutorMetrics`) and making the tracker part of the process itself, so that when it closes, it will complain for every un-freed allocation.

Reply:
This is now updated